src/sentry/coreapi.py

```diff
@@ -561,30 +543,17 @@ def _decode(self):
         # mutates data
         data = helper.validate_data(data)
 
-        if 'sdk' not in data:
-            sdk = helper.parse_client_as_sdk(auth.client)
-            if sdk:
-                data['sdk'] = sdk
-            else:
-                data['sdk'] = {}
-
-        data['sdk']['client_ip'] = self._client_ip
-
         data['project'] = self._project.id
         data['key_id'] = self._key.id
-
-        # we always fill in the IP so that filters and other items can
-        # access it (even if it eventually gets scrubbed)
-        helper.ensure_has_ip(
-            data,
-            self._client_ip,
-            set_if_missing=auth.is_public or
-            data.get('platform') in ('javascript', 'cocoa', 'objc')
-        )
+        data.setdefault('sdk', helper.parse_client_as_sdk(auth.client))
```

Comment:
The only opinion I have here is the rhs is being evaluated regardless if `sdk` is set or not. So even if it's set correctly, we're doing the unnecessary work from `parse_client_as_sdk`.

Reply:
fixed